sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -119,6 +119,13 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
         clusterBySpecOpt.getOrElse(ClusterBySpec(Nil)), conf.resolver))
       AlterTableSetPropertiesCommand(table.catalogTable.identifier, prop, isView = false)
 
+    case AlterTableCollation(ResolvedTable(catalog, _, table: V1Table, _), collation)
```

Comment:
Since this is a new feature, I think we can directly go with the v2 command. i.e. do not match this command here, and eventually Spark will plan it as `AlterTableExec`

Reply:
@cloud-fan, actually this exposed a bug in v2AlterTableCommands implementation of AlterTableCollation(), which was not setting the collation properly for existing tables. I have changed this code to use setProperty instead (similar to how other table catalog props are altered), and added a test case in v2/DescribeTableSuite to ensure that this path now works correctly.